libflux/go/libflux/parser_test.go

```diff
@@ -0,0 +1,21 @@
+// +build libflux
+
+package libflux_test
+
+import (
+	"testing"
+
+	"github.com/influxdata/flux/libflux/go/libflux"
+)
+
+func TestParse(t *testing.T) {
+	text := `
+package main
+
+from(bucket: "telegraf")
+	|> range(start: -5m)
+	|> mean()
+`
+	ast := libflux.Parse(text)
+	ast.Free()
```

Comment:
It is not clear to me why you `Free` again. Is that only to show that the operation is idempotent?

Because `Parse` sets the finalizer to free memory. That's my doubt.

Reply:
It's mostly because I didn't have anything to test. I wanted to show that the methods worked properly without panicking or, in the case of C code, it would segfault. Also, the garbage collector is unlikely to run before the test is over so the memory would not be freed. At the same time, the memory not being freed when the program ends is not a big deal. It's mostly the only reason finalizers exist.